modules/users/default.nix

```diff
@@ -245,10 +257,36 @@ in
             fi
 
             dscl . -create ${dsclUser} IsHidden ${if v.isHidden then "1" else "0"}
-            ${optionalString v.createHome "createhomedir -cu ${name}"}
           fi
-          # Always set the shell path, in case it was updated
-          dscl . -create ${dsclUser} UserShell ${lib.escapeShellArg (shellPath v.shell)}
+
+          # Always update all properties on users to keep them inline with configuration
+          dscl . -create ${dsclUser} PrimaryGroupID ${toString v.gid}
+          ${optionalString (v.description != null) "dscl . -create ${dsclUser} RealName ${lib.escapeShellArg v.description}"}
+          ${optionalString (v.shell != null) "dscl . -create ${dsclUser} UserShell ${lib.escapeShellArg (shellPath v.shell)}"}
```

Comment:
Is `dscl` not deprecated for these tasks? I’m a little worried about having this long‐term split where we’re mixing `sysadminctl` and `dscl` when they might behave differently.

Reply:
https://ss64.com/mac/sysadminctl.html

`sysadminctl` only deals with creating, deleting users, changing passwords and secure token settings, as far as I can tell updating settings is still `dscl`'s responsibility